src/OpenTelemetry.AutoInstrumentation/ContinuousProfiler/ContinuousProfilerProcessor.cs

```diff
@@ -86,13 +100,20 @@ private static void ActivityChanged(AsyncLocalValueChangedArgs<Activity?> sender
         NativeMethods.ContinuousProfilerSetNativeContext(0, 0, 0);
     }
 
-    private static void SampleReadingThread(BufferProcessor sampleExporter, TimeSpan exportInterval)
+    private void SampleReadingThread(BufferProcessor sampleExporter)
     {
         while (true)
         {
-            // TODO Graceful shutdown and Task.Delay https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/3216
-            Thread.Sleep(exportInterval);
+            // TODO Task.Delay https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/3216
+            var shutdownRequested = _shutdownTrigger.Wait(_exportInterval);
+
             sampleExporter.Process();
+
+            if (shutdownRequested)
+            {
+                Logger.Debug("Shutdown requested, exiting continuous profiler's exporter thread.");
+                return;
+            }
```

Comment:
I would put it before process. loosing one set of sampled data should not be a problem.
The bigger problem should be waiting for finishing the process.